### DISCUSSION TITLE
Auto disconnect when using a streaming software (SamCast - AAC format)

### DISCUSSION THREAD

**Discussion by Andrewunique** [AzuraCast/AzuraCast#5131] [Q&A]
### Installation Method

Docker Installation

### AzuraCast Release Channel

Rolling Release Channel

### Current AzuraCast Version

Rolling Release #6e739a1 (2022-02-01 21:32)

### What happened?

hello,
i am using AzuraCast through a reseller for 6 months. I use a streamer DJ for 99.9% of the time, but the AutoDJ(liquidsoap) is enabled for back-up purposes. As a broadcasting software i use SamCast, and the format used is AAC. The Dj/Streamer buffer is set to 20 sec in Azura panel just to be sure that there is a stable connection, without hiccups. So, practically the delay between the actual sound in the automation software and the radio stream is around 20 sec. But i noticed that this interval decreases by time, and in 3-4 days after the Streamer was started (the encoder in SamCast was started) this delay is around 8-9 sec (even if the buffer in Azura panel is still set on 20 sec). At this points, there is a moment when the streaming software (SamCast) disconnects for 20-30 seconds, the AutoDJ takes over, and the sounds from the external encoder (SamCast) returns after this 20-30sec. After this event, the delay between automation software and radio stream is again around 20 sec. This is happening on and on after every 3-4 days. It is like a buffer refresh or smth. Are you aware about this behavior? Maybe is related to the fact that AAC doesnt have a constant bitrate?
thank you!

**Comment by Vaalyn** [comment]
On the "Edit Station Profile" page in the "AutoDJ" tab you can configure the `DJ/Streamer Buffer Time (Seconds)`.
Your's is most likely set to 20 seconds, this means when a DJ connects Liquidsoap will first put 20s of data in it's buffer and then switch the stream over to the buffered DJ.

When the connection between the DJ and Liquidsoap has a lag or the DJ momentarily doesn't send data fast enough the buffer will start to deplete. This means you'll have for example 15s in the buffer instead of 20s.

Since the DJs streaming software is sending at the same speed as the stream pushes out it won't fill up the buffer again. It will stay at the lower value. If this happens multiple times until the buffer is completely empty the DJ will be disconnected by Liquidsoap and the DJs streaming software most likely automatically re-connects immediately. Then LS will first fill up the 20s buffer again and then switch back to the buffered DJ.

**Comment by Andrewunique** [comment]
Thanks a lot for your answer. Vaalyn,
Is it any solution for avoiding this situation? Is it related to aac format which doesnt have a constant bit rate?

**Reply by Vaalyn** [reply]
There probably isn't a solution.

As far as I'm aware this is not something related to the format you are using but rather the general streaming protocol that doesn't have any "hey DJ client, my buffer is not full please send more data until it's full again" feature.

**Comment by Andrewunique** [comment]
a, ok, i got it.
I used Sonic Panel until few months ago, and it didnt have this issue, but sonic panel doesnt support AAC @192 kbps for AutoDJ.
So i will stay with Azura for a better sound quality even if the stream will restart on every 4 days :)
Thanks a lot for this information!
